utils/setStatus.js

```diff
@@ -1,26 +1,27 @@
 'use strict';
 
 const statusCodes = require('http').STATUS_CODES
+    , isUndefined = require('./tools').isUndefined
     , setStatus = () => {
         return function (status) {
             /* eslint-disable no-invalid-this */
             const that = this;
             /* eslint-enable no-invalid-this */
-            let k;
 
-            if (status in statusCodes) {
+            if (!isUndefined(statusCodes[status])) {
                 that.statusCode = parseInt(status, 10);
                 that.statusMessage = statusCodes[status];
-                return;
+                return true;
             }
 
-            for (k in statusCodes) {
+            return Object.keys(statusCodes).some((k) => {
                 if (statusCodes[k].toLowerCase() === status.toLowerCase()) {
                     that.statusCode = parseInt(k, 10);
```

Comment:
To keep this code a little more side-effect free it would be nice to use the result of the `.some` in an if block that sets the statusCode and the statusMessage instead of doing them inline in the function.